modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/DeleteOneTree.pm

```diff
@@ -47,8 +47,17 @@ sub write_output {
         $tree->preload;
         $gene_tree_adaptor->delete_tree($tree);
         $tree->release_tree;
-    } );
+    }, 1, 2 );  # Retry once and wait 2 seconds between retries
+    $self->_check_clean_eradication();  # Make sure that the tree has been removed - a messy reindex is a pointless reindex
 }
 
-1;
+sub _check_clean_eradication {
+    my $self = shift @_;
 
+    my $gene_tree_adaptor = $self->compara_dba->get_GeneTreeAdaptor;
+    if (my $tree = $gene_tree_adaptor->fetch_by_dbID($self->param_required('gene_tree_id'))) {
```

Comment:
FYI this is fine for this adaptor, but other adaptors cache their data and further calls to `fetch_by_dbID` keep on returning the same object (unless the adaptor has a method to delete the object _and_ remove the entry from the cache). For instance, MLSS adaptor has a `delete` method that removes the entry from the table _and_ the cache, but not the GenomeDB adaptor, and the default `delete_by_dbID` method it inherits from BaseAdaptor is not cache-aware.

Reply:
I had not considered that at all, that's good to know, thanks!